packages/sdk/src/fetcher/core/index.ts

```diff
@@ -209,6 +219,133 @@ class Fetcher implements ICoreFetcher {
         }
         return contents;
     }
-}
 
-export const CoreFetcher = new Fetcher();
+    public async resolveTemplates({
+        ipfsGatewayURL,
+        templates,
+    }: ResolveTemplatesParams): Promise<ResolvedTemplate[]> {
+        return Promise.all(
+            templates.map(async (template) => {
+                const templateSpecificationCID = template.specificationCID;
+                const resolvedTemplateSpecificationCID = `${templateSpecificationCID}/base.json`;
+                const rawTemplateSpecification = (
+                    await this.fetchContentFromIPFS({
+                        ipfsGatewayURL,
+                        cids: [resolvedTemplateSpecificationCID],
+                    })
+                )[resolvedTemplateSpecificationCID];
+                if (!rawTemplateSpecification)
+                    throw new Error(
+                        `couldn't fetch template specification with cid ${templateSpecificationCID}`
+                    );
+                const parsedTemplateSpecification = JSON.parse(
+                    rawTemplateSpecification
+                );
+                const specification = new TemplateSpecification(
+                    templateSpecificationCID,
+                    parsedTemplateSpecification.name,
+                    parsedTemplateSpecification.description,
+                    parsedTemplateSpecification.tags,
+                    parsedTemplateSpecification.repository,
+                    parsedTemplateSpecification.commitHash
+                );
+                return ResolvedTemplate.from(template, specification);
+            })
+        );
+    }
+
+    protected async resolveEntity<T extends KPIToken | Oracle = KPIToken>({
+        ipfsGatewayURL,
+        entity,
+    }: {
+        entity: T;
+        ipfsGatewayURL: string;
+    }): Promise<T extends KPIToken ? ResolvedKPIToken : ResolvedOracle> {
+        const template = (
+            await this.resolveTemplates({
+                ipfsGatewayURL,
+                templates: [entity.template],
+            })
+        )[0];
+
+        if (!("oracles" in entity))
+            // FIXME: this is ugly but the ts compiler is complaining otherwise (not sure why)
```

Comment:
I think I'd remove this comment or make it shorter, in my opinion It's better just to have the condition.

Reply:
Not sure I'm quite getting it. The comment refers to the explicit cast done a line below.